pyblish/logic.py

```diff
@@ -354,9 +355,9 @@ def Iterator(plugins, context, state=None):
         "ordersWithError": set()
     }
 
-    # We'll add "default" target if no targets are registered. This happens
-    # when running the Iterator directly without registering any targets.
-    targets = registered_targets() or ["default"]
+    # Include "default" target when no targets are requested.
+    targets = targets or ["default"]
+    targets += registered_targets()
```

Comment:
Will this change the behavior to always including previously registered targets as opposed to them being explicits overrides to the `targets`? Before this made sense, since explicitly providing a new list of targets would override the registered targets, but now it won't...

Shouldn't the behavior be to only use whatever is passed along when something is passed along, and if `None` then to use the registered targets? That way, if you'd want to explicitly run without the currently registered targets you still can... and if you'd want to run *with* the currently registered targets you can do so explicitly:

```python
registered_targets = api.registered_targets()
pyblish.util.publish(targets=registered_targets + ["bar"])
```

Reply:
Right, this will be an issue indeed. Instead it should be something like:

```python
# Include "default" target when no targets are requested.
_targets = targets or ["default"]

if targets is None:
    _targets += registered_targets()

plugins = plugins_by_targets(plugins, _targets)
```

I should also add a unit test to ensure that registered targets are not used when explicit targets are passed! Will work on it

Thanks for the feedback!

Reply:
Code updated, I also updated the [test_publishing_explicit_targets_with_global](https://github.com/buddly27/pyblish-base/blob/f97bcd3893e60f3b3a51659590f729989cfc8ebe/tests/test_util.py#L259) unit test to ensure that explicit targets override registered targets.